### DISCUSSION TITLE
Callback before redirection

### DISCUSSION THREAD

**Discussion by rzhkjr** [remix-run/react-router#10988] [Proposals]
Add 2 props for <Navlink>/<link>, the first one would be boolean, and the second callback, to which the link path is passed, this would simplify the work with the redirection warning in the interior of the site,

Real usage scenario: I'm doing online tic tac toe, and when I'm online I need to warn the player if he goes to another page, he will lose the rating. I implemented this functionality on 'useNavigate()', but it would be cool if it were native, and together with the  <Navlink>/<link>features

P.S. I can do the implementation of this feature

**Comment by kiliman** [comment]
Will `unstable_useBlocker` handle your use case?

https://github.com/remix-run/react-router/tree/4475b56d2e18014acfb8ff5f6077ff27df98d7fe/examples/navigation-blocking

**Comment by rzhkjr** [comment]
I didn't see this in the documentation, I think it should solve my problem, thanks
